src/SYNFPAN.m

```diff
@@ -0,0 +1,379 @@
+ . . . d log(jlog,"Panel already loaded, skipping")
+ . . d log(jlog,"Calling LAB^ISIIMP12 to add panel")
+ . . n RESTA,RC
+ . . s (RESTA,RC)=""
+ . . ;i $g(DEBUG)=1 ZWRITE MISC 
+ . . S RESTA=$$LAB^ISIIMP12(.RC,.MISC)
+ . . d log(jlog,"Return from LAB^ISIIMP12 was: "_$g(RESTA))
+ . . ;i $g(DEBUG)=1 ZWRITE RESTA
+ . . ;i $g(DEBUG)=1 ZWRITE RC
+ . . if +$g(RESTA)=1 do  ;
+ . . . s @eval@("panels","status","loaded")=$g(@eval@("panels","status","loaded"))+1
+ . . . s @eval@("panels",SYNZI,"status","loadstatus")="loaded"
+ . . else  s @eval@("panels","status","errors")=$g(@eval@("panels","status","errors"))+1
+ . ;b
+ ;
+ if $get(args("debug"))=1 do  ;
+ . m jrslt("source")=@json
+ . m jrslt("args")=args
+ . m jrslt("eval")=@eval
+ m jrslt("labsStatus")=@eval@("labsStatus")
+ set jrslt("result","status")="ok"
+ set jrslt("result","loaded")=$g(@eval@("panels","status","loaded"))
+ set jrslt("result","errors")=$g(@eval@("labs","status","errors"))
+ m result("status")=jrslt("result")
+ q 0
+ ;
+ONELAB(MISCARY,json,ien,zj,jlog,eval,lablog)
+ ;
+ new obscode set obscode=$get(@json@("entry",ien,"resource","code","coding",1,"code"))
+ do log(lablog,"result "_zj_" code is: "_obscode)
+ do log(jlog,"result "_zj_" code is: "_obscode)
+ set @eval@("labs",SYNZI,"vars",zj_" code")=obscode
+ ;
+ ;
+ new codesystem set codesystem=$get(@json@("entry",ien,"resource","code","coding",1,"system"))
+ do log(jlog,"result "_zj_" code system is: "_codesystem)
+ do log(lablog,"result "_zj_" code system is: "_codesystem)
+ set @eval@("labs",SYNZI,"vars",zj_" codeSystem")=codesystem
+ ;
+ ; determine the value and units
+ ;
+ new value set value=$get(@json@("entry",ien,"resource","valueQuantity","value"))
+ if value="" d  ;
+ . new sctcode,scttxt
+ . s sctcode=$get(@json@("entry",ien,"resource","valueCodeableConcept","coding",1,"code"))
+ . s scttxt=$get(@json@("entry",ien,"resource","valueCodeableConcept","coding",1,"display"))
+ . s value=sctcode_"^"_scttxt
+ . d ADJUST(.value)
+ . do log(jlog,"result "_zj_" value is: "_value)
+ . do log(lablog,"result "_zj_" value is: "_value)
+ . set @eval@("labs",SYNZI,"vars",zj_" value")=value
+ . ;
+ . ;new unit set unit=$get(@json@("entry",SYNZI,"resource","valueQuantity","unit"))
+ . ;do log(jlog,"units are: "_unit)
+ . ;set @eval@("labs",SYNZI,"vars","units")=unit
+ . ;
+ . ; add to MISCARY
+ . ;
+ . n VLAB ; VistA lab name
+ . s VLAB=$$MAP^SYNQLDM(obscode,"labs")
+ . i VLAB="" d  quit
+ . . do log(jlog,"result "_zj_" VistA Lab not found for loinc="_obscode)
+ . . do log(lablog,"result "_zj_" VistA Lab not found for loinc="_obscode)
+ . d log(jlog,"result "_zj_" VistA Lab for "_obscode_" is: "_VLAB)
+ . d log(lablog,"result "_zj_" VistA Lab for "_obscode_" is: "_VLAB)
+ . s MISCARY("LAB_TEST",VLAB)=value
+ . ;
+ . d log(lablog,"Return from LAB^ISIIMP12 was: 1^Part of a Lab Panel "_SYNZI)
+ . s @eval@("labs",ien,"status","loadstatus")="loaded"
+ . s @eval@("labs","status","loaded")=$g(@eval@("labs","status","loaded"))+1
```

Comment:
After your changes, labs that are parts of panels still got loaded even after a panel is loaded.

Reply:
I fixed that.